modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -852,12 +880,16 @@ private Index maybeExecuteRollover(ClusterState state, DataStream dataStream, bo
                 ),
                 e
             );
-            DataStream latestDataStream = clusterService.state().metadata().getProject().dataStreams().get(dataStream.getName());
+            DataStream latestDataStream = clusterService.state()
+                .metadata()
+                .getProject(project.id())
+                .dataStreams()
+                .get(dataStream.getName());
             if (latestDataStream != null) {
```

Comment:
The existing code handles null by ignoring it. I think we should honour it by not throwing if the project is _not_ found. Not sure why the existing code did not use the ClusterState that is passed-in? Does it expect the latest state might be different from the one passed in?

Reply:
I indeed assumed it was to take intermediate cluster states into account, that's why I didn't change it to use the passed-in project. I'll update it to not throw when the project doesn't exist, good point :+1:.